dot/state/storage_notify.go

```diff
@@ -60,6 +60,16 @@ func (s *StorageState) RegisterStorageChangeChannel(sub StorageSubscription) (by
 	s.changedLock.Lock()
 	s.subscriptions[id] = &sub
 	s.changedLock.Unlock()
+	// notifyStorageSubscriptions here to send storage value of current state
+	sr, err := s.blockState.BestBlockStateRoot()
+	if err != nil {
+		logger.Debug("error registering storage change channel", "error", err)
+	}
+	go func() {
+		if err := s.notifyStorageSubscriptions(sr); err != nil {
+			logger.Warn("failed to notify storage subscriptions", "error", err)
+		}
+	}()
```

Comment:
why is this moved here? it seems like this would cause every subscription to be notified when a new subscription is registered, and only for the current state root?

Reply:
Right, I put this here so that subscriptions are notified when a new subscription is created so that the requester get's the current value when they subscribe.  It would cause all subscriptions to be notified except that subscriptions were values have not changed will not notify.  I'd like to find a cleaner way to do this, but calling notifyStorageSubscriptions seems like the easiest way to do this.  Other suggestions?

Reply:
maybe you could pull the logic that notifies one subscription from the `notifyStorageSubscriptions` function  into another function and call that for just this subscription?

Reply:
good idea, done.